modelskill/skill.py

```diff
@@ -299,8 +313,18 @@ def __repr__(self):
     def _repr_html_(self):
         return self.df._repr_html_()
 
-    def __getitem__(self, x):
-        return self.df[x]
+    def __getitem__(self, x) -> SkillArray | SkillFrame:
```

Comment:
Use `@overload` to improve the auto-completion experience.

Like this:
https://github.com/DHI/mikeio/blob/5e84e3c9c39dbbd76c84e813b547567aa4dcf773/mikeio/dataset/_dataset.py#L638-L648